From 4e489a956b6886cccb47a3682f3391b0ad59e8f0 Mon Sep 17 00:00:00 2001
From: David Wronek <david-w@ctemplar.com>
Date: Wed, 17 Nov 2021 14:59:51 +0100
Subject: [PATCH 199/210] arm64: dts: qcom: msm8916-huawei-y635: Add sensors

Huawei Y635 only has 2 sensors, all supported by existing kernel drivers:

1. Avago APDS9930 proximity/light sensor
2. STMicroelectronics LIS3DH accelerometer

Signed-off-by: David Wronek <david-w@ctemplar.com>
---
 .../boot/dts/qcom/msm8916-huawei-y635.dts     | 54 +++++++++++++++++++
 1 file changed, 54 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts b/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
index 1c6c6c12edc..b52faa9a0d9 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
@@ -34,6 +34,44 @@ volume-up {
 	};
 };
 
+&blsp_i2c2 {
+	status = "okay";
+
+	accelerometer@18 {
+		compatible = "st,lis3dh-accel";
+		reg = <0x18>;
+		interrupt-parent = <&msmgpio>;
+		interrupts = <115 IRQ_TYPE_EDGE_RISING>;
+
+		vdd-supply = <&pm8916_l17>;
+		vddio-supply = <&pm8916_l6>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&accel_int_default>;
+
+		mount-matrix =	"0", "-1", "0",
+				"-1", "0", "0",
+				"0", "0", "1";
+	};
+
+	proximity@39 {
+		compatible = "avago,apds9930";
+		reg = <0x39>;
+
+		interrupt-parent = <&msmgpio>;
+		interrupts = <113 IRQ_TYPE_EDGE_FALLING>;
+
+		vdd-supply = <&pm8916_l17>;
+		vddio-supply = <&pm8916_l6>;
+
+		led-max-microamp = <100000>;
+		amstaos,proximity-diodes = <1>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&prox_irq_default>;
+	};
+};
+
 &blsp1_uart2 {
 	status = "okay";
 };
@@ -187,6 +225,14 @@ l18 {
 };
 
 &msmgpio {
+	accel_int_default: accel-int-default {
+		pins = "gpio112";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
 	gpio_keys_default: gpio-keys-default {
 		pins = "gpio107";
 		function = "gpio";
@@ -194,4 +240,12 @@ gpio_keys_default: gpio-keys-default {
 		drive-strength = <2>;
 		bias-pull-up;
 	};
+
+	prox_irq_default: prox-irq-default {
+		pins = "gpio113";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
 };
-- 
2.34.1

